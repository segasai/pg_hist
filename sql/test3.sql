select * from pg_hist('select a from test1 where a is null',ARRAY[1], ARRAY[0],ARRAY[2]) 
    as (a int, b int);
select * from pg_hist_1d('select a from test1 where a is null',ARRAY[1], ARRAY[0],ARRAY[2]);
-- checks for out of hist cases
select * from pg_hist_1d('select -1000::real from test1',
            ARRAY[100],ARRAY[0],ARRAY[100]);
select * from pg_hist_1d('select -1000::double precision from test1',
            ARRAY[100],ARRAY[0],ARRAY[100]);
select * from pg_hist_1d('select -1000::int from test1',
            ARRAY[100],ARRAY[0],ARRAY[100]);
select * from pg_hist_1d('select -1000::bigint from test1',
            ARRAY[100],ARRAY[0],ARRAY[100]);
select * from pg_hist_2d('select -1000::bigint,-1000. from test1',
        ARRAY[100,100],ARRAY[0,0],ARRAY[100,100]);
select * from pg_hist_3d('select -1000::bigint,-1000.,-100::int from test1',
        ARRAY[100,100,100],ARRAY[0,0,0],ARRAY[100,100,100]);
select * from pg_hist_1d('select 1000 from test1',
            ARRAY[100],ARRAY[0],ARRAY[100]);
select * from pg_hist_2d('select 1000::bigint,1000. from test1',
        ARRAY[100,100],ARRAY[0,0],ARRAY[100,100]);
select * from pg_hist_3d('select 1000::bigint,1000.,100::int from test1',
        ARRAY[100,100,100],ARRAY[0,0,0],ARRAY[100,100,100]);
