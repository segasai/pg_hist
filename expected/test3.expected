 a | b 
---+---
   |  
(1 row)

 x1 | hist 
----+------
(0 rows)

 x1 | hist 
----+------
(0 rows)

 x1 | hist 
----+------
(0 rows)

 x1 | hist 
----+------
(0 rows)

 x1 | hist 
----+------
(0 rows)

 x1 | x2 | hist 
----+----+------
(0 rows)

 x1 | x2 | x3 | hist 
----+----+----+------
(0 rows)

 x1 | hist 
----+------
(0 rows)

 x1 | x2 | hist 
----+----+------
(0 rows)

 x1 | x2 | x3 | hist 
----+----+----+------
(0 rows)

